File: 2199-two-furthest-houses-with-different-colors/two-furthest-houses-with-different-colors.c
int maxDistance(int* colors, int colorsSize){
    int maxDist = 0;
    for(size_t a = 0;a < colorsSize;a++){
        for(size_t b = 0;b < colorsSize;b++){
            if(a == b || colors[a] == colors[b]){continue;}
            int distCalc = abs(a-b);
            if(distCalc > maxDist){maxDist = distCalc;}
        }
    }
    return maxDist;
}

File: 2199-two-furthest-houses-with-different-colors/two-furthest-houses-with-different-colors.cs
public class Solution {
    public int MaxDistance(int[] colors) {
        int maxDist = 0;
        for(int i = 0; i < colors.Length; i++){
            for (int j = 0; j < colors.Length; j++){
                if(colors[i] != colors[j]){
                    maxDist = Math.Max(maxDist,Math.Abs(i-j));
                }
            }
        }
        return maxDist;
    }
}